controller/src/main/java/org/jboss/as/controller/AbstractOperationContext.java

```diff
@@ -1583,7 +1638,39 @@ private void rollbackAddedServices() {
                 }
             }
         }
+    }
+
+    /** Lightweight AbstractStep that only aims to populate the response's OUTCOME and ROLLED_BACK nodes in Stage.DONE */
+    private class DoneStagePlaceholder extends AbstractStep {
+
+        private final ModelNode outcome;
+        private final ModelNode rolledBack;
+        private final boolean failed;
+
+        private DoneStagePlaceholder(Step replaces) {
+            this.predecessor = replaces.predecessor;
+            // Cache the OUTCOME and ROLLED_BACK nodes and the failed status
+            // in order to populate the nodes in finalizeStep.
+            // We don't need any of the other response data.
+            this.outcome = replaces.response.get(OUTCOME);
+            this.rolledBack = replaces.response.get(ROLLED_BACK);
```

Comment:
This will create an undefined "rolled-back" node in the response even if there is no roll-back

Reply:
Thanks so much; this was a big flaw.

AbstractOperationContext L898-905 fixes it. 

https://github.com/wildfly/wildfly-core/pull/5540/files#diff-ff4d8c15cef21b2291ac6c3e80c9ae63ae4d32fa8e554ee376d8a3a3fa761336R898-R905

The updates to AbstractGlobalOperationsTestCase and AbstractControllerTestBase test this. Without the fix ^^^ 136 tests failed once I made the test changes.